src/Common/Process.cpp

```diff
@@ -119,11 +128,30 @@ std::wstring Process::OutputOf( const std::wstring& rCommandLine, int& rResult )
 
 		return Output;
 	}
+#else
+#define READ_CHUNK_LEN 1024 // TODO: This is kind of arbitrary?
+
+	int fds [ 2 ];
+	pipe( fds );
+	fcntl( fds [ 0 ], F_SETFL, O_NONBLOCK ); // Don't want to block on read
```

Comment:
can't we use select() or something similar to see if we have data available? Assuming that is better than marking the file descriptor as non-blocking.

Reply:
Not really why its set to nonblocking - we know that the process is terminated by the time the `read` executes so polling would be unnecesary - the nonblock is to read arbitrary data into a dynamic buffer instead of allocating a randomly sized static buffer which could overflow